src/test/basics/Buffer_test.cpp

```diff
@@ -0,0 +1,240 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github0.com/ripple/rippled
+    Copyright (c) 2012-2016 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/basics/Buffer.h>
+#include <ripple/beast/unit_test.h>
+#include <cstdint>
+
+namespace ripple {
+namespace test {
+
+struct Buffer_test : beast::unit_test::suite
+{
+    bool sane (Buffer const& b)
+    {
+        if (b.size() == 0)
+            return b.data() == nullptr;
+
+        return b.data() != nullptr;
+    }
+
+    std::uint8_t
+    generate()
+    {
+        static std::uint8_t x = 7;
+        return x++;
+    }
+
+    void run()
+    {
+        std::uint8_t const data[] =
+        {
+            0xa8, 0xa1, 0x38, 0x45, 0x23, 0xec, 0xe4, 0x23,
+            0x71, 0x6d, 0x2a, 0x18, 0xb4, 0x70, 0xcb, 0xf5,
+            0xac, 0x2d, 0x89, 0x4d, 0x19, 0x9c, 0xf0, 0x2c,
+            0x15, 0xd1, 0xf9, 0x9b, 0x66, 0xd2, 0x30, 0xd3
+        };
+
+        Buffer b0;
+        BEAST_EXPECT (sane (b0));
+        BEAST_EXPECT (b0.empty());
+
+        Buffer b1 { 0 };
+        BEAST_EXPECT (sane (b1));
+        BEAST_EXPECT (b1.empty());
+        std::memcpy(b1.alloc(16), data, 16);
+        BEAST_EXPECT (sane (b1));
+        BEAST_EXPECT (!b1.empty());
+        BEAST_EXPECT (b1.size() == 16);
+
+        Buffer b2 { b1.size() };
+        BEAST_EXPECT (sane (b2));
+        BEAST_EXPECT (!b2.empty());
+        BEAST_EXPECT (b2.size() == b1.size());
+        std::memcpy(b2.data(), data + 16, 16);
+
+        Buffer b3 { data, sizeof(data) };
+        BEAST_EXPECT (sane (b3));
+        BEAST_EXPECT (!b3.empty());
+        BEAST_EXPECT (b3.size() == sizeof(data));
+        BEAST_EXPECT (std::memcmp (b3.data(), data, b3.size()) == 0);
```

Comment:
IMNSHO, making these `Buffer const` would make the remaining test cases more solid. I pushed a commit for you if you want to use it. https://github.com/ximinez/rippled/commit/4c5cfb9a443dda5f68e87b162045fb5dff186e42